src/decoding.cc

```diff
@@ -189,18 +267,34 @@ namespace ctranslate2 {
     StorageView attention_step_device(dtype, device);
 
     StorageView coverage;
-
+    std::unique_ptr<BiasedDecoder> biased_decoder;
     for (dim_t step = start_step; step < max_step; ++step) {
       // Compute log probs for the current step.
       decoder(step,
               topk_ids.to(device),
               state,
-              &logits,
+              &logits,  // output shape: (cur_batch_size*beam_size x 1 x vocab_size), if not expanded beam_size is 1
               (attention || _coverage_penalty != 0) ? &attention_step_device : nullptr);
-      ops::LogSoftMax()(logits, log_probs);
+
+      if (bias_towards_prefix) {
+        if (!biased_decoder) {
+          biased_decoder.reset(new BiasedDecoder());
```

Comment:
We try to follow this pattern for unique_ptr:

```suggestion
          biased_decoder = std::make_unique<BiasedDecoder>();
```